app/over_react_redux/todo_client/lib/src/components/create_input.dart

```diff
@@ -14,26 +15,31 @@ UiFactory<CreateInputProps> CreateInput =
 
 @Props(keyNamespace: '') // No namespace so prop forwarding works when passing to the JS TextField component.
 class _$CreateInputProps extends UiProps {
-  void Function(String s) onCreate;
+  @requiredProp void Function(String s) onCreate;
   bool autoFocus;
   String label;
   String placeholder;
 }
 
 @Component2()
 class CreateInputComponent extends UiComponent2<CreateInputProps> {
+  @visibleForTesting
+  final textFieldRef = createRef().jsRef;
```

Comment:
@greglittlefield-wf it doesn't seem to work that way when the ref we want is nested within the JS component we're wrapping - and that JS component is a function component that makes use of `React.forwardRef`.

Reply:
Okay, I dug into this, and it looks like the problem isn't actually because the JS component uses forwardRef, but rather just because it's using a ref from the Dart code.

When you pass a ref as some other prop (in this case, `'inputRef'`) react-dart doesn't convert it, so when the JS component goes to put that ref on something else, it fails.

So, it's needed here, but it wouldn't be needed if you were just setting a `'ref'` as is the case in display_list.dart.